### ISSUE TITLE
Prerained model of SVR

### ISSUE THREAD

**Issue by gxu-tz** [czq142857/BSP-NET-pytorch#8]
Thank you for your code.I have seen other issues that the pretrained model is without overlap loss and code of generator class in modelSVR.py corresponds to Phase 1.
So to produce better results for shape reconstruction,should I retrain the AE model on phase 0 → phase 3 and  change the code of generator class in modelSVR.py and then retrain the SVR model?

**Comment by czq142857** [comment]
Yes. Phase 3 provides slightly better results. But to truly improve the result quality, you may want to sample more points on the shapes (in the data pre-processing step), because the current data with at most 16384 sampled points (inside+outside) per shape is not sufficient for good reconstructions.

**Comment by gxu-tz** [comment]
Thank you for the response.
1.Does sampling more points on the shapes in the data pre-processing step mean changing the code
`batch_size_3 = 16 * 16 * 16 * 4` in 2_gather_256vox_16_32_64.py in IM-NET repository?
2.I have another question. BSP-NET can generate sharp edges, so can it only generate approximate polygons for circles, as shown in Figure 7 in the paper？

**Comment by czq142857** [comment]
1. Yes, and also batch_size_2.

2. Yes. One solution is to use quadric surfaces to replace planes, but it may lead to overfitting. See the following papers for references.
(a) https://arxiv.org/abs/2106.14274
(b) https://arxiv.org/abs/2104.05652
(c) Code of (a): https://github.com/czq142857/BSP-NET-original/tree/master/bsp_fc_qs_deeper_2048q

**Comment by gxu-tz** [comment]
Thank you for your prompt reply,it helps me a lot!I will pay attention to other work of your team.
